src/context/AppProvider/AppContext.tsx

```diff
@@ -135,85 +135,69 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
           const adapter = chainAdapterManager.get(chainId)
           if (!adapter) continue
 
-          switch (chainId) {
-            case ethChainId: {
-              if (!supportsETH(wallet)) continue
-              const pubkey = await adapter.getAddress({ wallet })
-              if (!pubkey) continue
-              acc.push({ [chainId]: pubkey.toLowerCase() })
-              break
-            }
-            case avalancheChainId: {
-              if (!supportsEthSwitchChain(wallet)) continue
-              const pubkey = await adapter.getAddress({ wallet })
-              if (!pubkey) continue
-              acc.push({ [chainId]: pubkey.toLowerCase() })
-              break
-            }
-            case btcChainId: {
+          const { chainNamespace, chainReference } = fromChainId(chainId)
+
+          switch (chainNamespace) {
+            case CHAIN_NAMESPACE.Bitcoin: {
               if (!supportsBTC(wallet)) continue
+
+              const coin = {
+                [CHAIN_REFERENCE.BitcoinMainnet]: 'bitcoin',
```

Comment:
adapter.getCoinName()?

Reply:
https://github.com/shapeshift/web/pull/2280/commits/8c1f4042d3807963c1fc1adbb6141f175295a97e

Reply:
Or use the newly added `bitcoinAssetMap` etc from https://github.com/shapeshift/lib/pull/914/files#diff-9623b77b3cfaf7a86b7dfae686e16b9306e4d03d019e563d7b61d348630b0078R39-R43 ? 

Not sure if it makes sense to write a method similar to `adapter.getDisplayName()` only for the purpose of making a hdwallet's `getPublicKeys()` compliant string